src/Compilers/Core/Portable/Syntax/InternalSyntax/SyntaxDiagnosticInfoList.cs

```diff
@@ -110,27 +103,26 @@ public bool MoveNext()
 
             private void PushNodeOrToken(GreenNode node)
             {
-                var token = node as SyntaxToken;
-                if (token != null)
+                if (node.IsToken)
```

Comment:
Nope.  PushNodeOrToken is only called if node != null.  I can add an assert to that effect though!